src/os/desktop.zig

```diff
@@ -85,3 +85,30 @@ pub fn desktopEnvironment() DesktopEnvironment {
         else => .other,
     };
 }
+
+pub const DisplayType = enum {
+    macos,
+    other,
+    wayland,
+    windows,
+    x11,
+};
+
+/// Detect what desktop environment we are running under. This is mainly used on
+/// Linux to enable or disable GTK client-side decorations but there may be more
+/// uses in the future.
+pub fn displayType() DisplayType {
+    return switch (comptime builtin.os.tag) {
+        .macos => .macos,
+        .windows => .windows,
+        .linux => dt: {
+            if (@inComptime()) @compileError("Checking for the display type on Linux must be done at runtime.");
```

Comment:
@jcollie *suggestion*: Should word this as follows: "On Linux, the display type must be determined dynamically at runtime."